v2/api/compute/customizations/vm_extensions.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package customizations
+
+import (
+	"github.com/go-logr/logr"
+
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/core"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+)
+
+var _ extensions.ErrorClassifier = &VirtualMachineExtension{}
+
+// ClassifyError evaluates the provided error, returning whether it is fatal or can be retried.
+func (e *VirtualMachineExtension) ClassifyError(
+	cloudError *genericarmclient.CloudError,
+	apiVersion string,
+	log logr.Logger,
+	next extensions.ErrorClassifierFunc) (core.CloudErrorDetails, error) {
+	details, err := next(cloudError)
+	if err != nil {
+		return core.CloudErrorDetails{}, err
+	}
+
+	// It's weird to retry on OperationNotAllowed as it certainly sounds like a fatal error, but
+	// it primarily happens for quota errors on VM/VMSS, which we do want to retry on as the quota may free
+	// up at some point in the future.
```

Comment:
Are there other resources with quota where this might be needed? 

I don't think the requisite things in Batch are resources, but maybe other providers?

Reply:
The problem is that the setting of this code is going to be specific to individual RPs. Not every RP will use `OperationNotAllowed` for "no quota". Compute does, so making this check specific to them for now makes sense. In the future if we have specific codes from other RPs that are quota that are reported to us (or that we find) we can expand to have custom error classifiers for those RPs that check for their specific `code`, but it's not universal across all RPs so I think we need to go RP by RP.